fileserver/cook/file_server.py

```
@@ -0,0 +1,164 @@
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+#  IN THE SOFTWARE.
+#
+"""Module implementing a file server to serve Cook job logs. """
+
+import os
+from operator import itemgetter
+from pathlib import Path
+from stat import *
+
+import gunicorn.app.base
+from flask import Flask, jsonify, request, send_file
+from gunicorn.six import iteritems
+
+app = Flask(__name__)
+
+
+class FileServerApplication(gunicorn.app.base.BaseApplication):
+
+    def __init__(self, options=None):
+        self.options = options or {}
+        self.application = app
+        super(FileServerApplication, self).__init__()
+
+    def load_config(self):
+        config = dict([(key, value) for key, value in iteritems(self.options)
+                       if key in self.cfg.settings and value is not None])
+        for key, value in iteritems(config):
+            self.cfg.set(key.lower(), value)
+
+    def load(self):
+        return self.application
+
+
+@app.route('/files/download')
+@app.route('/files/download.json')
+def download():
+    path = request.args.get('path')
+    if path is None:
+        return "Expecting 'path=value' in query.\n", 400
+    if not os.path.exists(path):
+        return "", 404
+    if os.path.isdir(path):
+        return "Cannot download a directory.\n", 400
+    return send_file(path, as_attachment=True)
+
+
+def try_parse_int(param_name, val):
+    err_message = lambda: f"Failed to parse {param_name}: Failed to convert '{val}' to number.\n"
+    try:
+        int_val = int(val)
+    except ValueError as _:
+        return (None, err_message())
+    except Exception as _:
+        return (None, err_message())
+    return (int_val, None)
```

Nitpick: It's considered more Pythonic to throw the error and catch it rather than returning a pair.